src/services/encryptedStorage.js

```diff
@@ -70,6 +78,22 @@ EncryptedStorage.prototype.save = async function (id: string, data: Object, forc
   return null;
 };
 
+EncryptedStorage.prototype.migrateFromLocalStorage = async function (storageData: Object) {
+  const { wallet = {} } = get(storageData, 'wallet', {});
+  if (isEmpty(wallet)) return Promise.resolve();
+
+  try {
+    printLog('Encrypted Storage migrating data');
+    // eslint-disable-next-line i18next/no-literal-string
+    await this.save('wallet', { wallet });
+    // eslint-disable-next-line i18next/no-literal-string
+    await storage.remove('wallet');
+  } catch (e) {
+    reportErrorLog('AsyncStorage migration to Encrypted Storage failed', { error: e });
```

Comment:
If this fails - what happens?

Reply:
@lalitkumarrock I know sorry but why?

Reply:
I just need to know what is the fallback scenario here

Reply:
Here failure chances are very low. if fails then open the onboarding screen.
I pushed one changes. [de04d01](https://github.com/pillarwallet/pillarwallet/pull/3488/commits/de04d0106f81c56b86d736433000c01877986647)